app/pages/nonsipp/common/IdentityTypePage.scala

```diff
@@ -86,8 +86,7 @@ case class IdentityTypePage(srn: Srn, index: Max5000, identitySubject: IdentityS
       case (Some(IdentityType.UKCompany), _) => removePages(userAnswers, pagesFirstPart(srn))
       case (Some(IdentityType.UKPartnership), _) => removePages(userAnswers, pagesFirstPart(srn))
       case (Some(IdentityType.Other), _) => removePages(userAnswers, pagesFirstPart(srn))
-      case (Some(_), _) => Try(userAnswers)
-      case (None, _) => removePages(userAnswers, pages(srn))
```

Comment:
@cyeniceri (FYI @westwater) I think this line was handling removal of all details when the entity is removed (on UI delete link on the list page). Could you please double check if this still works?